gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerFragment.java

```diff
@@ -147,12 +152,40 @@ private void onMapReady(MapAdapter map) {
     homeScreenViewModel
         .getBottomSheetState()
         .observe(this, state -> onBottomSheetStateChange(state, map));
-    binding.addFeatureBtn.setOnClickListener(
+    binding.mapControls.addFeatureBtn.setOnClickListener(
         __ -> homeScreenViewModel.onAddFeatureBtnClick(map.getCameraTarget()));
+    binding.moveFeature.confirmButton.setOnClickListener(
+        __ -> showConfirmationDialog(map.getCameraTarget()));
+    binding.moveFeature.cancelButton.setOnClickListener(__ -> cancelRepositionMode());
     enableLocationLockBtn();
     mapContainerViewModel.getMbtilesFilePaths().observe(this, map::addTileOverlays);
   }
 
+  private void showConfirmationDialog(Point point) {
+    new Builder(getContext())
+        .setTitle(R.string.move_point_confirmation)
+        .setPositiveButton(android.R.string.ok, (dialog, which) -> moveToNewPosition(point))
+        .setNegativeButton(android.R.string.cancel, (dialog, which) -> cancelRepositionMode())
+        .setCancelable(true)
+        .create()
+        .show();
+  }
+
+  private void moveToNewPosition(Point point) {
+    mapContainerViewModel
+        .getSelectedFeature()
+        .ifPresentOrElse(
+            feature -> {
+              homeScreenViewModel.updateFeature(feature, point);
+              setDefaultMode();
+            },
+            () -> Timber.e("Feature is null, can't move to new position"));
```

Comment:
A better error might be "No feature selected", since it described _why_ we might be in this state.